src/utils/logger/logger-typescript.ts

```diff
@@ -164,15 +171,27 @@ export const loadTypeScriptDiagnostic = (tsDiagnostic: Diagnostic) => {
   return d;
 };
 
-const flattenDiagnosticMessageText = (tsDiagnostic: Diagnostic, diag: string | DiagnosticMessageChain | undefined) => {
+/**
+ * Flatten a TypeScript diagnostic object into a string which can be easily
+ * included in a Stencil diagnostic record.
+ *
+ * @param tsDiagnostic a TypeScript diagnostic record
+ * @param diag a {@link DiagnosticMessageChain} or a string with further info
+ * @returns a string with the relevant error message
+ */
+const flattenDiagnosticMessageText = (
+  tsDiagnostic: Diagnostic,
+  diag: string | DiagnosticMessageChain | undefined
+): string => {
   if (typeof diag === 'string') {
     return diag;
   } else if (diag === undefined) {
     return '';
   }
 
   const ignoreCodes: number[] = [];
-  const isStencilConfig = tsDiagnostic.file.fileName.includes('stencil.config');
+  // `tsDiagnostic.file` can be `undefined`, so we need to be a little careful here
+  const isStencilConfig = (tsDiagnostic?.file?.fileName ?? '').includes('stencil.config');
```

Comment:
Can `tsDiagnostic` be `undefined` here? Do we need to update the function signature here?

Reply:
oh I don't think so...probably I can remove the first `?.` 😅